cuda/kernels.cu

```diff
@@ -127,6 +139,28 @@ __global__ void organize_points(Point *d_points, int *d_categories, Point *bl,
 	}
 }
 
+__global__ void quadrant_search(Query *queries, int num_queries,
+								QuadrantBoundary *boundaries,
+								int num_boundaries, int *results) {
+	int idx = blockIdx.x * blockDim.x + threadIdx.x;
+	if (idx < num_queries) {
+		Query query = queries[idx];
+		int result = -1;
+
+		for (int i = 0; i < num_boundaries; i++) {
```

Comment:
If we sort our boundaries based on a comparator. We can apply binary search here to reduce the time.. if time is becomg an issue?

Reply:
Binary search would cause thread divergence. We want all threads to do the same operation as much as possible.

Reply:
how will it cause thread divergence? All threads will be doing Binary search...

Reply:
The GPU will run the warp for threads that meet a certain criteria together, basically the if condition would be executed together. Imagine all 32 threads in the warp have a separate boundary that they are a part of. In this case, the warp will execute the same code 32 times, because all the threads have a certain different number of times to execute. Binary search is not a good algorithm for GPU